superset-frontend/src/components/DatabaseSelector/DatabaseSelector.test.tsx

```diff
@@ -0,0 +1,225 @@
+            'created_by.last_name': 'Created By Last Name',
+            database_name: 'Database Name',
+            explore_database_id: 'Explore Database Id',
+            expose_in_sqllab: 'Expose In Sqllab',
+            force_ctas_schema: 'Force Ctas Schema',
+            id: 'Id',
+          },
+          list_columns: [
+            'allow_csv_upload',
+            'allow_ctas',
+            'allow_cvas',
+            'allow_dml',
+            'allow_multi_schema_metadata_fetch',
+            'allow_run_async',
+            'allows_cost_estimate',
+            'allows_subquery',
+            'allows_virtual_table_explore',
+            'backend',
+            'changed_on',
+            'changed_on_delta_humanized',
+            'created_by.first_name',
+            'created_by.last_name',
+            'database_name',
+            'explore_database_id',
+            'expose_in_sqllab',
+            'force_ctas_schema',
+            'id',
+          ],
+          list_title: 'List Database',
+          order_columns: [
+            'allow_csv_upload',
+            'allow_dml',
+            'allow_run_async',
+            'changed_on',
+            'changed_on_delta_humanized',
+            'created_by.first_name',
+            'database_name',
+            'expose_in_sqllab',
+          ],
+          result: [
+            {
+              allow_csv_upload: false,
+              allow_ctas: false,
+              allow_cvas: false,
+              allow_dml: false,
+              allow_multi_schema_metadata_fetch: false,
+              allow_run_async: false,
+              allows_cost_estimate: null,
+              allows_subquery: true,
+              allows_virtual_table_explore: true,
+              backend: 'postgresql',
+              changed_on: '2021-03-09T19:02:07.141095',
+              changed_on_delta_humanized: 'a day ago',
+              created_by: null,
+              database_name: 'examples',
+              explore_database_id: 1,
+              expose_in_sqllab: true,
+              force_ctas_schema: null,
+              id: 1,
+            },
+          ],
+        },
+      } as any;
+    },
+  );
+
+  defaultProps = {
+    dbId: 1,
+    formMode: false,
+    isDatabaseSelectEnabled: true,
+    readOnly: false,
+    schema: 'public',
+    sqlLabMode: true,
+    getDbList: jest.fn(),
+    getTableList: jest.fn(),
+    handleError: jest.fn(),
+    onDbChange: jest.fn(),
+    onSchemaChange: jest.fn(),
+    onSchemasLoad: jest.fn(),
+    onChange: jest.fn(),
+  };
+});
+
+test('Should render', async () => {
+  const props = { ...defaultProps };
+  await act(async () => {
+    render(<DatabaseSelector {...props} />);
+  });
+  expect(screen.getByTestId('DatabaseSelector')).toBeInTheDocument();
+});
+
+test('Refresh should works', async () => {
+  const props = { ...defaultProps };
+  await act(async () => {
+    render(<DatabaseSelector {...props} />);
+  });
+
+  expect(SupersetClientGet).toBeCalledTimes(2);
```

Comment:
Maybe use `waitFor` here? 

```
render(<DatabaseSelector {...props} />);

await waitFor(() => {
   expect(SupersetClientGet).toBeCalledTimes(2);
   ...
});

userEvent.click(screen.getByRole('button'));

await waitFor(() => {
   expect(SupersetClientGet).toBeCalledTimes(3);
   ...
});


```

Reply:
In this case the results are different. I want to guarantee the exact number of calls.
```
await waitFor(() => {
   expect(SupersetClientGet).toBeCalledTimes(2); // this is valid even when calling 2 or more times
});
```
```
expect(SupersetClientGet).toBeCalledTimes(2); // this is valid even when calling only 2 times
```

The intention here is to ensure that calls are being made exactly this number of times.